### ISSUE TITLE
Add protection for AI directory leaks

### ISSUE THREAD

**Issue by fzipi** [coreruleset/coreruleset#4474]
### Description

Modern times. We need to protect people from leaking AI token, secrets and others.

We should add at least these:

```
# AI coding assistant directories
.claude/
.cursor/
.continue/
.copilot/
.aider/
```

Source: https://ironpeak.be/blog/leaking-secrets-from-the-claud/

### How to reproduce the misbehavior (-> curl call)

```bash

```

### Your Environment

* CRS Latest

### Confirmation

- [x] I have removed any personal data (email addresses, IP addresses, passwords, domain names) from any logs posted.

**Comment by etiennemunnich** [comment]
Happy to help with this. I'll dig into this more.

**Comment by fzipi** [comment]
Thanks @etiennemunnich. It would be nice if it makes the next LTS ;)

**Comment by etiennemunnich** [comment]
Ack! Been working this and have a question; would a typical approach to add this as PL2 then moved to PL1 later?

**Comment by fzipi** [comment]
I think this should complement lists that we already have. We are probably matching literal strings, so it would be very difficult to have `.claude/..` in a natural language phrase.

**Comment by fzipi** [comment]
So: PL1.

**Comment by etiennemunnich** [comment]
Proposed change:

- Add rule 930140 (PL1 block) and ai-critical-artifacts.data file.
- Insert rule in REQUEST-930-APPLICATION-ATTACK-LFI.conf after 930130.

References (supporting links for ai-critical-artifacts.data entries):

Dot-folders:
  .claude/    https://code.claude.com/docs/en/settings  https://code.claude.com/docs/en/hooks
  .cursor/    https://docs.cursor.com/context/rules-for-ai
  .continue/  https://docs.continue.dev/customize/deep-dives/configuration   <---- This is an agentic coding tool but concerned about the FPs this might generate.
  .aider/     https://aider.chat/docs/config.html
  .roo/       https://docs.roocode.com/features/mcp/using-mcp-in-roo
  .zed/       https://zed.dev/docs/reference/all-settings
  .cline/     https://docs.cline.bot/customization/cline-rules  https://docs.cline.bot/mcp/adding-and-configuring-servers
  .kiro/      https://kiro.dev/docs/getting-started/first-project  https://kiro.dev/docs/hooks
  .windsurf/  https://docs.windsurf.com
  .rovodev/   https://support.atlassian.com/rovo/docs/set-custom-instructions-for-code-reviews  https://support.atlassian.com/rovo/docs/manage-rovo-dev-cli-settings/
  .codex/     https://developers.openai.com/codex/config-basic
  .opencode/  https://opencode.ai/docs/config/
  .a0proj/    https://www.agent-zero.ai/p/docs/projects/  https://github.com/agent0ai/agent-zero
  .plandex/   https://plandex.ai (plans, context)
  .fabric/    https://github.com/danielmiessler/fabric (patterns, fabric.env)

There are a couple other AI related projects we could consider, like OpenClaw/ClawCode, Openhands, OpenUI, n8n etc etc as well.

If we're happy with this list let me know and I'll add to the branch for review.

**Comment by fzipi** [comment]
I'm happy with the list 😄 It will be ever-growing, like anything AI these days...

**Comment by etiennemunnich** [comment]
That's for sure!  Will submit shortly.
